crates/solvers/src/domain/dex/mod.rs

```diff
@@ -105,7 +105,7 @@ impl Swap {
         self,
         order: order::Order,
         gas_price: auction::GasPrice,
-        sell_token: auction::Price,
+        sell_token: Option<auction::Price>,
```

Comment:
This feels a little icky... You need a `sell_token: Price` for certain orders, but not for others. Open to ideas for refactoring the interface here to make it more coherent.

Reply:
Which orders need this, and which don't? Is there a rule to it?

Reply:
Yes there is:
- Orders with solver-computed fees need this
- Orders with protocol-determined fees do not

Reply:
This is notably an issue with quoting, where the synthetic auction that is passed to the solver engine does not have reference prices (and can never have reference prices).